### ISSUE TITLE
Incorrect query for spending_by_practice endpoint

### ISSUE THREAD

**Issue by inglesp** [ebmdatalab/openprescribing#873]
Incorrect SQL is generated for spending_by_practice when the query...

 * is for a product or presentation
 * covers multiple organisations
 * includes a date

There is a test for this in f81d688d, and a fix in 98920090.

I'm fairly sure that our JS code never calls this endpoint with a date.  (See [chart_utils.js](https://github.com/ebmdatalab/openprescribing/blob/master/openprescribing/media/js/src/chart_utils.js#L26-L84).)

But somebody has been, and moreover has been for multiple organisations:

```
$ grep spending_by_practice nginx-access.log | grep date | grep -E 'org=\w+,' | wc -l
2402
```

(nginx-access.log goes back to 2018-05-13.)

But nobody has (recently) made any API calls that satisfy all three bullet point conditions:

```
$ grep spending_by_practice nginx-access.log | grep date | grep -E 'org=\w+,' | grep -E 'code=\w{10,}' | wc -l
0
```

We should review and deploy the fix ASAP.  And we should think about how to communicate this to users of the API.

**Comment by inglesp** [comment]
Looking at the recent logs I get the feeling that somebody has been trying to reverse engineer the API -- there are lots of queries for things that the API will ignore, eg `date=2017-12-01&date=2017-01-02`.

And the requests initially come from the browser, while later they come from localhost, and then they come from http://healthcaredataviz.azurewebsites.net (eg [this page](http://healthcaredataviz.azurewebsites.net/ViewFocusGroup.html?org=C82099,C82614,C82643,C82020)).

Would be worth getting in touch with whoever runs that site (probably related to www.healthcaredataviz.com) to find out about their API usage.

**Comment by inglesp** [comment]
This was fixed in #851.
